src/sm64ap.cpp

```diff
@@ -79,8 +79,12 @@ void SM64AP_RecvItem(int64_t idx, bool notify) {
         case SM64AP_ID_CANNONUNLOCK(0) ... SM64AP_ID_CANNONUNLOCK(15-1):
             sm64_have_cannon[idx-(SM64AP_ID_CANNONUNLOCK(0))] = true;
             break;
-        case SM64AP_ID_ABILITY(0) ... SM64AP_ID_ABILITY(SM64AP_NUM_ABILITIES-1):
-            sm64_have_abilities[idx-(SM64AP_ID_ABILITY(0))] = true;
+        case SM64AP_ID_ABILITY(0):
+            sm64_have_abilities[idx - SM64AP_ABILITY_OFFSET + 1] = sm64_have_abilities[idx - SM64AP_ABILITY_OFFSET];
+            sm64_have_abilities[idx - SM64AP_ABILITY_OFFSET] = true;
+            break;
+        case SM64AP_ID_ABILITY(1) ... SM64AP_ID_ABILITY(SM64AP_NUM_ABILITIES-1):
+            sm64_have_abilities[idx-SM64AP_ABILITY_OFFSET] = true;
```

Comment:
Spacing

Reply:
Fixed in 9746f720e2f38a8bdb6961b735559cd705ac381f